integration/spark/src/main/spark3.1/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala

```diff
@@ -129,9 +129,32 @@ class CarbonHelperSqlAstBuilder(conf: SQLConf,
     val properties: Map[String, String] = getPropertyKeyValues(tablePropertyList)
     val tableProperties = convertPropertiesToLowercase(properties)
 
+    var partitionTransformList: Seq[Transform] = Seq()
+    var partitionByStructFields: Seq[StructField] = Seq()
     // validate partition clause
-    val partitionByStructFields = Option(partitionColumns).toSeq
-        .flatMap(x => visitPartitionFieldList(x)._2)
+    var partitionTransformList: Seq[Transform] = Seq()
+    var partitionByStructFields: Seq[StructField] = Seq()
+    Option(partitionColumns).toSeq.foreach(x => {
+      val (t, s) = visitPartitionFieldList(x)
+      partitionTransformList = partitionTransformList ++ t
+      partitionByStructFields = partitionByStructFields ++ s
+    })
```

Comment:
Option(partitionColumns).toSeq.foreach(x => {
      val (t, s) = visitPartitionFieldList(x)
      partitionTransformList = partitionTransformList ++ t
      partitionByStructFields = partitionByStructFields ++ s
    })  
does this no need to use foreach, the input of visitPartitionFieldList is PartitionFieldListContext type and after func visitPartitionFieldList the result is two list.

Reply:
Have made the change to directly call the `visitPartitionFieldList `function.